contracts/strategies/BaseStrategy.sol

```diff
@@ -38,6 +38,20 @@ abstract contract BaseStrategy is IBaseStrategy {
         _STRATEGY_ID = keccak256(abi.encode(_strategyName));
     }
 
+    /// ===============================
+    /// ========= Initialize ==========
+    /// ===============================
+
+    /// @notice Initialize the strategy
+    /// @param __poolId The pool id
+    /// @param _data custom data to be used to initialize the strategy
+    function initialize(uint256 __poolId, bytes memory _data) external virtual override {
```

Comment:
i'd add a `@dev` tag commenting that `__BaseStrategy_init` is `onlyAllo`, and that it cannot be called twice